src/wget_agent/agent/wget_agent.c

```diff
@@ -1156,13 +1162,82 @@ char* GetVersionControlCommand(int withPassword)
   else if (0 == strcmp(GlobalType, Type[1]))
   {
     replace_url_with_auth();
+
+  strcpy(temp1, GlobalParam);
+  branchFound = strstr(temp1, "--branch");
+
+  if(branchFound) 
+  {
+    tokens = strtok(branchFound, " ");
+    while(tokens != NULL) 
+    {
+      if(i == 1) 
+      {
+        branchFound = tokens;
+        break;
+      }
+      tokens = strtok(NULL, " ");
+      i++;
+    }
+    i=0; 
+    tokens = "";
+  }
+
+  strcpy(temp2, GlobalParam);
+
+  commitFound = strstr(temp2, "--commit");
+  if(commitFound) 
+  {
+    tokens = strtok(commitFound, " ");
+    while(tokens != NULL) 
+    {
+      if(i == 1)
+      {
+        commitFound = tokens;
+      }
+      tokens = strtok(NULL, " ");
+      i++;
+    } 
+  }
+
+
     if (GlobalProxy[0] && GlobalProxy[0][0])
     {
-      res = asprintf(&command, "git config --global http.proxy %s && git clone %s %s %s  && rm -rf %s/.git", GlobalProxy[0], GlobalURL, GlobalParam, tmpfile_dir, tmpfile_dir);
+      if(commitFound == NULL && branchFound == NULL)
+      {
+      res = asprintf(&command, "git config --global http.proxy %s && git clone %s %s && cd %s && git checkout %s && rm -rf %s/.git", GlobalProxy[0], GlobalURL, tmpfile_dir, tmpfile_dir, GlobalParam, tmpfile_dir);
+      }
+      else if(commitFound == NULL && branchFound != NULL)
+      {
+        res = asprintf(&command, "git config --global http.proxy %s && git clone %s %s && cd %s && git checkout %s && rm -rf %s/.git", GlobalProxy[0], GlobalURL, tmpfile_dir, tmpfile_dir, branchFound, tmpfile_dir);
```

Comment:
`GlobalParam` initially had `--single-branch` and `--branch` to reduce the `git checkout`s. Is it possible to keep the same behavior? Otherwise the `git clone` will download all data and then you need to perform 2 checkouts if there is a commit provided.